python/pyspark/sql/streaming/query.py

```diff
@@ -618,12 +618,23 @@ def addListener(self, listener: StreamingQueryListener) -> None:
 
         .. versionadded:: 3.4.0
 
+        .. versionchanged:: 3.5.0
+            Supports Spark Connect.
+
         Parameters
         ----------
         listener : :class:`StreamingQueryListener`
             A :class:`StreamingQueryListener` to receive up-calls for life cycle events of
             :class:`~pyspark.sql.streaming.StreamingQuery`.
 
+        Notes
+        -----
+        This function behaves differently in Spark Connect mode.
+        In Connect, the provided functions doesn't have access to variables defined outside of it.
```

Comment:
Is this a limitation for Python that variables outside of the functions cannot be accessed?

Reply:
Yes, similar to python foreachBatch, because now the code is running on the server, so if you define a variable outside of the function, even if it will be picked by pickle, any modification to that variable won't be sent back to client. ForeachBatch side might have a clearer doc for this:
https://github.com/apache/spark/pull/42096/files#r1269909249